libarchive/ffi.py

```diff
@@ -240,5 +243,7 @@ def ffi(name, argtypes, restype, errcheck=None):
     c_int, check_int)
 ffi('write_finish_entry', [c_archive_p], c_int, check_int)
 
+ffi('write_fail', [c_archive_p], c_int, check_int)
+
```

Comment:
It's not used directly by any of context managers in python-libarchive-c, but we use it in our project to allow cancelling creation of archive. Since this library is a convenient wrapper around libarchive we think that it's the best place for it (and make this wrapper more complete).
Turns out that if you want to discard the archive you were creating, you can not just stop creating it. Context managers will close the archive, and libarchive will try to fill the missing contents with 000000000000000000..., and that can cause memory overflows. You have to mark the archive as failed (by that function) to tell libarchive that it should give up any work on it.